app/src/features/surveys/components/SurveyFundingSourceForm.tsx

```diff
@@ -73,59 +77,82 @@ const SurveyFundingSourceForm = () => {
       <FieldArray
         name="funding_sources"
         render={(arrayHelpers: FieldArrayRenderProps) => (
-          <Box>
-            {values.funding_sources.map((surveyFundingSource: ISurveyFundingSource, index: number) => {
-              return (
-                <Box
-                  mb={3}
-                  display="flex"
-                  gap={2}
-                  alignItems="flex-start"
-                  key={surveyFundingSource.survey_funding_source_id}>
-                  <AutocompleteField
-                    id={`funding_sources.[${index}].funding_source_id`}
-                    name={`funding_sources.[${index}].funding_source_id`}
-                    label="Funding Source"
-                    sx={{ flex: 6 }}
-                    options={fundingSources.map((fundingSource) => ({
-                      value: fundingSource.funding_source_id,
-                      label: fundingSource.name
-                    }))}
-                    loading={fundingSourcesDataLoader.isLoading}
-                    required
-                  />
-                  <DollarAmountField
-                    label="Amount (Optional)"
-                    id={`funding_sources.[${index}].amount`}
-                    name={`funding_sources.[${index}].amount`}
-                    value={surveyFundingSource.amount}
-                    onChange={handleChange}
-                    sx={{ flex: 4 }}
-                  />
-                  <Box mt={1}>
-                    <IconButton
-                      data-testid={`funding-form-delete-button-${index}`}
-                      title="Remove Funding Source"
-                      aria-label="Remove Funding Source"
-                      onClick={() => arrayHelpers.remove(index)}
-                      sx={{ ml: -1 }}>
-                      <Icon path={mdiTrashCanOutline} size={1} />
-                    </IconButton>
-                  </Box>
-                </Box>
-              );
-            })}
+          <Stack gap={1}>
+            <TransitionGroup
+              component={Stack}
+              gap={1}
+              role="list"
+              sx={{
+                '&:not(:has(div[role=listitem]))': {
+                  display: 'none'
+                }
+              }}>
+              {values.funding_sources.map((surveyFundingSource: ISurveyFundingSource, index: number) => {
+                return (
+                  <Collapse role="listitem" key={index}>
```

Comment:
Same here.
https://dev.to/shiv1998/why-not-to-use-index-as-key-in-react-lists-practical-example-3e66